blockchain/chain_sync/src/tipset_syncer.rs

```diff
@@ -53,41 +50,33 @@ use state_tree::StateTree;
 const MAX_TIPSETS_TO_REQUEST: u64 = 100;
 
 #[derive(Debug, Error)]
-pub enum TipsetProcessorError {
+pub enum TipsetProcessorError<C: Consensus> {
     #[error("TipsetRangeSyncer error: {0}")]
-    RangeSyncer(#[from] TipsetRangeSyncerError),
+    RangeSyncer(#[from] TipsetRangeSyncerError<C>),
     #[error("Tipset stream closed")]
     StreamClosed,
     #[error("Tipset has already been synced")]
     AlreadySynced,
 }
 
 #[derive(Debug, Error)]
-pub enum TipsetRangeSyncerError {
-    #[error("Tipset added to range syncer does share the same epoch and parents")]
-    InvalidTipsetAdded,
+pub enum TipsetRangeSyncerError<C: Consensus> {
+    // #[error("Tipset added to range syncer does share the same epoch and parents")]
+    // InvalidTipsetAdded,
```

Comment:
Do we really want to keep this variant commented?

Reply:
I don't know, I just saw it wasn't used and tried commenting out to make sure. Wanted to check the other ones as well but there were too many; I'll have another go. If you don't think it's needed I'll remove it now.